dask/dataframe/rolling.py

```diff
@@ -111,9 +114,12 @@ def _call_method(self, method_name, *args, **kwargs):
         if self.rolling_kwargs['axis'] in [0, 'rows']:
             # roll in the partition direction (will need to access neighbor)
             for i in range(1, self.obj.npartitions + 1):
+                # Get just the needed values from the previous partition
+                window = self.rolling_kwargs['window']
+                prior_chunk_tail = (tail, (old_name, i-1), window-1)
```

Comment:
Thanks for the thorough explanation. I hadn't properly appreciated that each call wasn't a task.